trin-execution/src/types/trie_proof.rs

```diff
@@ -8,6 +8,15 @@ pub struct TrieProof {
     pub proof: Vec<Bytes>,
 }
 
+impl TrieProof {
+    pub fn iter(trie_proof: TrieProof) -> impl Iterator<Item = TrieProof> {
+        (0..trie_proof.proof.len()).map(move |i| Self {
+            path: trie_proof.path[0..i].to_vec(),
```

Comment:
I don't think it is so simple as you will have extension nodes on the way, which contribute more than one nibble to the path... I think we should leverage TrieWalker, as I believe it already takes this into consideration.

Reply:
I forgot thank you for reminding me I will fix it